sys/random/Makefile

```diff
@@ -11,4 +14,28 @@ ifneq (,$(filter prng_tinymt32,$(USEMODULE)))
   DIRS += tinymt32
 endif
 
+ifeq (,$(RIOT_CI_BUILD))
+  # random seed value newly generated on each build
+  RANDOM_SEED := $(shell $(RIOTTOOLS)/randhex/randhex.py 32)
```

Comment:
Tbh this feature looks like a huge foot gun.
It will provide randomness during development, but once deployed all boards will behave the same.

If this feature is needed for a manual deployment where each node is flashed with a new build, it could be enabled through a pseudo-module. (But even then it would be much better to store the seed in a separate config region on the flash or EEPROM).

But making it default gives a false sense of randomness.

e.g. a user uses random() to decide when nodes should send something, during development with heterogeneous firmware versions this might work as expected.

When deployed all boards will send at the same time.

Can we instead work on improving the random API?

 - one 'true' random function (can be seeded by `puf_sram`, hwrng, …) that is not stable across reboots
 - instantiate randomness (seeded by CPU_ID or application) that can be used to generate 'stable' (across reboots) IDs or application behavior.

Reply:
100% my thoughts.

Could we compromise on making this non-default?

Reply:
Ping @nmeum, please comment.

Reply:
I think I commented on this deficiency in this PR already. I don't really have the time to work on this anyhow anymore. I wouldn't mind if this just gets closed, the issues with the random module laid out in this PR still stand though.